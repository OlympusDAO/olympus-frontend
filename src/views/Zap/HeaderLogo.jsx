import { Avatar, Box, SvgIcon } from "@material-ui/core";

const avatarStyle = { height: "35px", width: "35px", marginInline: "-4px" };
const iconStyle = { height: "35px", width: "35px", marginInline: "-4px", zIndex: 1 };
const viewBox = "0 0 32 32";
export default function HeaderLogo({ images, icons, avatarStyleOverride }) {
  if (images && !images.length) {
    images = [
      "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48.png",
      "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0x6b175474e89094c44da98b954eedeac495271d0f.png",
      "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0x0000000000000000000000000000000000000000.png",
    ];
  }
  return (
    <Box display="flex" marginBottom={2} flexDirection="row">
      {images?.map(image => (
        <Avatar src={image} style={avatarStyleOverride ?? avatarStyle} />
      ))}

      {icons?.map(icon => (
        <SvgIcon component={icon} viewBox={viewBox} style={iconStyle}></SvgIcon>
      ))}
    </Box>
  );
}
