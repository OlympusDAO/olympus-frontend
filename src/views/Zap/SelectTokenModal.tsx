import { t, Trans } from "@lingui/macro";
import {
  Avatar,
  Box,
  Button,
  CircularProgress,
  Dialog,
  DialogTitle,
  List,
  ListItem,
  ListItemAvatar,
  ListItemText,
  Paper,
  SvgIcon,
  Typography,
} from "@mui/material";
import { Token } from "@olympusdao/component-library";
import { trim } from "src/helpers";
import { ZapperToken } from "src/hooks/useZapTokenBalances";

import { ReactComponent as XIcon } from "../../assets/icons/x.svg";

function SelectTokenModal(
  handleClose: () => void,
  modalOpen: boolean,
  isTokensLoading: boolean,

  handleSelectToken: { (token: string): void },
  zapperCredit: JSX.Element,
  tokens: {
    regularTokens?: { [key: string]: ZapperToken };
    output?: boolean;
  },
) {
  return (
    <Dialog
      onClose={handleClose}
      open={modalOpen}
      keepMounted
      fullWidth
      maxWidth="xs"
      id="zap-select-token-modal"
      className="zap-card"
    >
      <DialogTitle>
        <Box display="flex" flexDirection="row" alignItems="center" justifyContent="space-between">
          <Box />
          <Box paddingLeft={6}>
            <Typography id="migration-modal-title" variant="h6" component="h2">
              {t`Select Token`}
            </Typography>
          </Box>
          <Button onClick={handleClose}>
            <SvgIcon component={XIcon} color="primary" />
          </Button>
        </Box>
      </DialogTitle>
      <Box paddingX="36px" paddingBottom="16px" paddingTop="12px">
        {isTokensLoading ? (
          <Box display="flex" justifyItems="center" flexDirection="column" alignItems="center">
            <CircularProgress />
            <Box height={24} />
            <Typography>
              <Trans>Dialing Zapper...</Trans>
            </Typography>
          </Box>
        ) : Object.entries(tokens).length == 0 ? (
          <Box display="flex" justifyContent="center">
            <Typography>
              <Trans>Ser, you have no assets...</Trans>
            </Typography>
          </Box>
        ) : (
          <Paper style={{ maxHeight: 300, overflow: "auto", borderRadius: 10 }}>
            <List>
              {tokens.regularTokens &&
                Object.entries(tokens.regularTokens)
                  .filter(token => !token[1].hide)
                  .sort((tokenA, tokenB) => tokenB[1].balanceUSD - tokenA[1].balanceUSD)
                  .map(token => (
                    <ListItem button onClick={() => handleSelectToken(token[0])} key={token[1].symbol}>
                      <ListItemAvatar>
                        <Avatar src={token[1].tokenImageUrl} />
                      </ListItemAvatar>
                      <ListItemText primary={token[1].symbol} />
                      <Box flexGrow={10} />
                      <ListItemText
                        style={{ display: "flex", flexDirection: "column", alignItems: "flex-end" }}
                        primary={`$${trim(token[1].balanceUSD, 2)}`}
                        secondary={trim(token[1].balance, 4)}
                      />
                    </ListItem>
                  ))}
              {tokens.output && (
                <>
                  <ListItem button onClick={() => handleSelectToken("sOHM")} key={"sOHM"}>
                    <ListItemAvatar>
                      <Token name={"sOHM"} />
                    </ListItemAvatar>
                    <ListItemText primary={"sOHM"} />
                  </ListItem>
                  <ListItem button onClick={() => handleSelectToken("gOHM")} key={"gOHM"}>
                    <ListItemAvatar>
                      <Token name={"wsOHM"} />
                    </ListItemAvatar>
                    <ListItemText primary={"gOHM"} />
                  </ListItem>
                </>
              )}
            </List>
          </Paper>
        )}
        {zapperCredit}
      </Box>
    </Dialog>
  );
}

export default SelectTokenModal;
