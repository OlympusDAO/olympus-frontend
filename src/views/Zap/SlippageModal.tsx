import { Trans } from "@lingui/macro";
import { Box, Dialog, DialogTitle, FormControl, makeStyles, Typography } from "@material-ui/core";
import { Input, PrimaryButton, SecondaryButton } from "@olympusdao/component-library";
import { SetStateAction, useEffect, useState } from "react";
import { trim } from "src/helpers";

const useStyles = makeStyles(theme => ({
  root: {
    [theme.breakpoints.down("sm")]: {
      paddingInline: "16px",
    },
    [theme.breakpoints.up("sm")]: {
      paddingInline: "64px",
    },
  },
}));

function SlippageModal(
  handleClose: () => void,
  modalOpen: boolean,
  currentSlippage: string,
  setCustomSlippage: { (value: SetStateAction<string>): void; (arg0: string): void },
  zapperCredit: JSX.Element,
) {
  const classes = useStyles();
  const [proposedSlippage, setProposedSlippage] = useState(currentSlippage);
  const [errorState, setErrorState] = useState<string | null>(null);
  const handleChangeProposedSlippage = (slippage: string) => {
    try {
      const slippageNumber = Number(slippage);
      setProposedSlippage(slippage);
      if (100 > slippageNumber && slippageNumber > 0) {
        if (slippageNumber < 1) {
          setErrorState("Lower slippage than recommended may cause transaction to fail");
        } else {
          setErrorState(null);
        }
      } else {
        setErrorState("Slippage must be between 0 and 100");
      }
    } catch (e) {
      console.error(e);
      setErrorState("Enter a valid slippage percentage");
    }
  };
  useEffect(() => handleChangeProposedSlippage(currentSlippage), [modalOpen]);
  const presetSlippageOptions = ["2.0", "3.0", "4.0"];
  return (
    <Dialog
      onClose={handleClose}
      open={modalOpen}
      fullWidth
      maxWidth="xs"
      id="zap-select-token-modal"
      className="zap-card"
    >
      <DialogTitle>
        <Box display="flex" flexDirection="row" alignItems="center" justifyContent="space-between">
          <Box />
          <Box paddingLeft={6}>
            <Typography id="migration-modal-title" variant="h6" component="h2">
              <Trans>Adjust Slippage</Trans>
            </Typography>
          </Box>
          <PrimaryButton icon="x" template="text" onClick={handleClose} />
        </Box>
      </DialogTitle>
      <Box paddingBottom="36px" className={classes.root}>
        <Typography color="textSecondary">
          <Trans>Important: Recommended slippage is 1-3% to avoid a failed transaction.</Trans>
        </Typography>
      </Box>
      <Box paddingBottom="16px" className={classes.root}>
        {/* <Paper style={{ maxHeight: 300, overflow: "auto", borderRadius: 10 }}> */}
        <FormControl className="slippage-input" variant="outlined" color="primary" size="small">
          <Input
            id="slippage"
            type="number"
            placeholder="Slippage Tolerance"
            value={proposedSlippage}
            onChange={e => handleChangeProposedSlippage(e.target.value)}
            endString="%"
          />
        </FormControl>

        <Box display="flex" flexDirection="row" justifyContent="space-between">
          {presetSlippageOptions.map(slippage => (
            <SecondaryButton size="small" onClick={() => handleChangeProposedSlippage(slippage)}>
              <Typography>{`${slippage}%`}</Typography>
            </SecondaryButton>
          ))}
        </Box>
        <Box paddingY="16px">{errorState ? <Typography color="error">{errorState}</Typography> : null}</Box>
        <Box display="flex" justifyContent={"center"}>
          <PrimaryButton
            size="small"
            color="primary"
            disabled={errorState != null}
            onClick={() => {
              if (errorState != null) {
                return;
              }
              setCustomSlippage(trim(+proposedSlippage, 1));
              handleClose();
            }}
          >
            <Typography>Adjust Slippage</Typography>
          </PrimaryButton>
        </Box>
        {zapperCredit}
      </Box>
    </Dialog>
  );
}

export default SlippageModal;
