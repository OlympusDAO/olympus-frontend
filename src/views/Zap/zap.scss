#zap-view {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  overflow: hidden;
}

#zap-select-token-modal {
  .MuiListItemText-primary,
  .MuiListItemText-secondary {
    text-align: right;
  }
}

#olyzaps-cta {
  .cta-box {
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
  }
  .icons-box {
    display: flex;
    align-items: center;
  }
  .button-box {
    display: flex;
  }
  .cta-header {
    font-size: 1rem;
    line-height: 1.24rem;
    padding-bottom: 0.5rem;
    font-weight: 400;
  }
  .cta-subheader {
    font-weight: 400;
  }
}

#olyzaps-info {
  .oly-info-card {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    padding: 0;
  }
  .oly-info-header-box {
    padding-bottom: 1.5rem;
  }
  .oly-info-body-header {
    padding-bottom: 0.5rem;
  }
  .oly-info-body {
    line-height: 1.25rem;
  }
  .button-box {
    display: flex;
    width: 100%;
    justify-content: center;
    align-items: center;
    padding-top: 2rem;
  }
  .learn-more-button {
    svg {
      margin-left: 4px;
      vertical-align: middle;
      font-size: 1.1em;
      height: 1.15rem;
      width: 1.15rem;
    }
    &:hover {
      svg {
        color: inherit;
      }
    }
  }
}

.stake-action-area {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  width: 100%;
  justify-content: space-around;
  margin-top: 10px;
}

.zap-input {
  width: 100%;
  margin-block: 20px;
  min-height: 70px;
}

.zap-stake-button {
  min-width: 140px !important;
  height: 43px !important;
}

.stake-wallet-notification {
  text-align: center;
  .wallet-menu {
    text-align: center;
    .connect-button {
      max-width: 300px !important;
      min-width: 240px !important;
      margin-bottom: 15px;
    }
  }
}
