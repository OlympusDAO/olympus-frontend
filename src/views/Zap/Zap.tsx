import "./Zap.scss";

import { Trans } from "@lingui/macro";
import { Box, Typography, Zoom } from "@material-ui/core";
import { Paper } from "@olympusdao/component-library";
import React, { useMemo } from "react";
import { useHistory } from "react-router";
import ConnectButton from "src/components/ConnectButton/ConnectButton";
import { useAppSelector } from "src/hooks";
import { usePathForNetwork } from "src/hooks/usePathForNetwork";
import { useWeb3Context } from "src/hooks/web3Context";

import ZapInfo from "./ZapInfo";
import ZapStakeAction from "./ZapStakeAction";

const Zap: React.FC = () => {
  const { address, networkId } = useWeb3Context();
  const history = useHistory();
  usePathForNetwork({ pathName: "zap", networkID: networkId, history });

  const tokens = useAppSelector(state => state.zap.balances);
  const inputTokenImages = useMemo(
    () =>
      Object.entries(tokens)
        .filter(token => token[0] !== "sohm")
        .map(token => token[1].tokenImageUrl)
        .slice(0, 3),
    [tokens],
  );

  return (
    <div id="zap-view">
      <Zoom in={true}>
        <Paper headerText={address && `Zap`}>
          <div className="staking-area">
            {!address ? (
              <div className="stake-wallet-notification">
                <div className="wallet-menu" id="wallet-menu">
                  <ConnectButton />
                </div>
                <Typography variant="h6">
                  <Trans>Connect your wallet to use Zap</Trans>
                </Typography>
              </div>
            ) : (
              <Box className="stake-action-area">
                <ZapStakeAction />
              </Box>
            )}
          </div>
        </Paper>
      </Zoom>
      <Zoom in={true}>
        <ZapInfo tokens={inputTokenImages} address={address} />
      </Zoom>
    </div>
  );
};

export default Zap;
