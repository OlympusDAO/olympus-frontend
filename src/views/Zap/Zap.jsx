import { useState, useRef, useEffect, useMemo } from "react";
import { Box, Button, Fade, Paper, Tab, Tabs, Typography, Zoom } from "@material-ui/core";
import TabPanel from "../../components/TabPanel";
import "./zap.scss";
import { useWeb3Context } from "src/hooks/web3Context";
import ZapStakeAction from "./ZapStakeAction";
import ZapInfo from "./ZapInfo";
import { useAppSelector } from "src/hooks";
import { Trans } from "@lingui/macro";
import { usePathForNetwork } from "src/hooks/usePathForNetwork";
import { useHistory } from "react-router";

function Zap() {
  const { address, connect } = useWeb3Context();
  const history = useHistory();
  const networkId = useAppSelector(state => state.network.networkId);
  usePathForNetwork({ pathName: "zap", networkID: networkId, history });

  const tokens = useAppSelector(state => state.zap.balances);
  const inputTokenImages = useMemo(
    () =>
      Object.entries(tokens)
        .filter(token => token[0] !== "sohm")
        .map(token => token[1].img)
        .slice(0, 3),
    [tokens],
  );

  return (
    <div id="zap-view">
      <Zoom in={true}>
        <Paper className="ohm-card">
          <div className="staking-area">
            {!address ? (
              <div className="stake-wallet-notification">
                <div className="wallet-menu" id="wallet-menu">
                  <Button variant="contained" color="primary" className="connect-button" onClick={connect} key={1}>
                    <Trans>Connect Wallet</Trans>
                  </Button>
                </div>
                <Typography variant="h6">
                  <Trans>Connect your wallet to use Zap</Trans>
                </Typography>
              </div>
            ) : (
              <>
                <Box className="stake-action-area">
                  <Box alignSelf="center" minWidth="420px" width="80%"></Box>
                  <ZapStakeAction />
                </Box>
              </>
            )}
          </div>
        </Paper>
      </Zoom>
      <Zoom in={true}>
        <ZapInfo tokens={inputTokenImages} address={address} />
      </Zoom>
    </div>
  );
}

export default Zap;
