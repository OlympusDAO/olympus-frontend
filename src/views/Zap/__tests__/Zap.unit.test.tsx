import { render } from "../../../testUtils";
import Zap from "../Zap";

describe("<Zap/>", () => {
  it("should render component", () => {
    const { container } = render(<Zap />);
    expect(container).toMatchSnapshot();
  });
});
