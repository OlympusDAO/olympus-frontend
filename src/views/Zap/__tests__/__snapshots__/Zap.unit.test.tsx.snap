// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Zap/> should render component 1`] = `
<div>
  <div
    id="zap-view"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 Paper-root  css-m19e0g-MuiPaper-root"
      style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1y6dg9a-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-159wrwf-MuiGrid-root"
        >
          <div
            class="staking-area"
          >
            <div
              class="stake-wallet-notification"
            >
              <div
                class="wallet-menu"
                id="wallet-menu"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-disableElevation MuiButtonBase-root Button-root undefined css-1s3qiyy-MuiButtonBase-root-MuiButton-root"
                  style="font-size: 1.2857rem;"
                  tabindex="0"
                  type="button"
                >
                  Connect Wallet
                </button>
              </div>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-rg0wkx-MuiTypography-root"
              >
                Connect your wallet to use Zap
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 ohm-card css-1c9lk12-MuiPaper-root"
      id="olyzaps-info"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-1bxbb5o-MuiGrid-root"
      >
        <div
          class="ZapInfo-infoBox MuiGrid-root ZapInfo-infoBoxItem MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-c8aaa2-MuiGrid-root"
        >
          <div
            class="ZapInfo-infoHeader oly-info-header-box MuiBox-root css-1m5be6d"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-1xhj18k"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1djwm8g-MuiSvgIcon-root"
                  focusable="false"
                  style="z-index: 1; margin-bottom: 16px;"
                  viewBox="0 0 32 32"
                >
                  <circle
                    cx="16"
                    cy="16"
                    fill="#F5AC37"
                    r="16"
                  />
                  <path
                    d="M16.59 17.13h6.08c.13 0 .19 0 .2-.17.05-.619.05-1.24 0-1.86 0-.12-.06-.17-.19-.17h-12.1c-.15 0-.19.05-.19.19v1.78c0 .23 0 .23.24.23h5.96Zm5.6-4.28a.195.195 0 0 0 0-.14 3.96 3.96 0 0 0-.36-.63 5.103 5.103 0 0 0-.74-.93 2.451 2.451 0 0 0-.461-.45 7.187 7.187 0 0 0-3-1.52 7.458 7.458 0 0 0-1.7-.18h-5.37c-.15 0-.17.06-.17.19v3.55c0 .15 0 .19.19.19h11.54s.1-.02.12-.08h-.049Zm0 6.36a2.336 2.336 0 0 0-.51 0H10.59c-.15 0-.2 0-.2.2v3.47c0 .16 0 .2.2.2h5.12c.245.02.49.002.73-.049a7.626 7.626 0 0 0 2.17-.48c.253-.088.498-.203.73-.34h.07a6.496 6.496 0 0 0 2.79-2.809s.07-.151-.01-.19v-.002ZM8.38 24.88v-5.53c0-.13 0-.15-.16-.15H6.05c-.12 0-.17 0-.17-.16v-1.9H8.2c.13 0 .18 0 .18-.17v-1.88c0-.12 0-.149-.16-.149H6.05c-.12 0-.17 0-.17-.16v-1.76c0-.11 0-.14.16-.14h2.15c.15 0 .19 0 .19-.19v-5.39c0-.16 0-.2.2-.2h7.5c.544.022 1.085.082 1.62.18a9.775 9.775 0 0 1 3.13 1.16 8.82 8.82 0 0 1 1.76 1.36c.396.412.753.857 1.07 1.33.314.48.575.994.78 1.53a.26.26 0 0 0 .3.21h1.79c.23 0 .23 0 .24.221v1.64c0 .16-.06.2-.221.2h-1.38c-.14 0-.18 0-.17.18.054.61.054 1.22 0 1.83 0 .17 0 .19.19.19h1.58c.07.09 0 .18 0 .27.01.117.01.234 0 .35v1.21c0 .17-.05.22-.2.22h-1.89a.25.25 0 0 0-.29.19 7.997 7.997 0 0 1-2.101 3.06c-.34.307-.697.595-1.07.86-.4.23-.79.47-1.2.66-.756.34-1.548.591-2.36.75a12.3 12.3 0 0 1-2.34.19h-6.94v-.01l.002-.002Z"
                    fill="#FEFEFD"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1djwm8g-MuiSvgIcon-root"
                  focusable="false"
                  style="margin-left: -6px; z-index: 1; margin-bottom: 16px;"
                  viewBox="0 0 32 32"
                >
                  <circle
                    cx="16"
                    cy="16"
                    fill="#fff"
                    r="15"
                    stroke="url(#wETH_svg__a)"
                    stroke-width="2"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M16.25 20.976 10 17.349 16.25 26V26l6.253-8.65-6.253 3.626Z"
                    fill="#708B96"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="m16.25 6 6.248 10.186-6.248-2.793L10 16.186 16.25 6Zm0 7.395L10 16.185l6.25 3.629 6.248-3.628-6.248-2.791Z"
                    fill="#424242"
                    fill-rule="evenodd"
                  />
                  <defs>
                    <lineargradient
                      gradientUnits="userSpaceOnUse"
                      id="wETH_svg__a"
                      x1="16"
                      x2="16"
                      y1="0"
                      y2="32"
                    >
                      <stop
                        stop-color="#444243"
                      />
                      <stop
                        offset="1"
                        stop-color="#708B96"
                      />
                    </lineargradient>
                  </defs>
                </svg>
              </div>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1m5binv-MuiTypography-root"
            >
              You Give
            </p>
          </div>
          <div
            class="ZapInfo-infoBody MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 oly-info-body-header css-69sme-MuiTypography-root"
            >
              Zap is a swap
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft oly-info-body css-z19mi8-MuiTypography-root"
            >
              A zap swap is a series of smart contracts that deploys one asset to another protocol to handle a trusted transaction.
            </p>
          </div>
        </div>
        <div
          class="ZapInfo-infoBox MuiGrid-root ZapInfo-infoBoxItem MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-c8aaa2-MuiGrid-root"
        >
          <div
            class="ZapInfo-infoHeader oly-info-header-box MuiBox-root css-1m5be6d"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1djwm8g-MuiSvgIcon-root"
              focusable="false"
              style="margin-bottom: 16px;"
              viewBox="0 0 32 32"
            >
              <circle
                cx="16"
                cy="16"
                fill="#8977F5"
                r="16"
              />
              <path
                clip-rule="evenodd"
                d="M 24.041 13.382 C 24.39 13.382 24.652 13.556 24.826 13.905 C 25.001 14.167 24.913 14.516 24.739 14.778 L 16.02 25.251 C 15.845 25.513 15.584 25.6 15.322 25.6 C 15.235 25.6 15.06 25.6 14.973 25.513 C 14.624 25.425 14.363 24.989 14.45 24.64 L 15.235 18.618 L 8.346 18.618 C 7.998 18.618 7.736 18.444 7.562 18.182 C 7.387 17.833 7.475 17.484 7.649 17.222 L 16.368 6.749 C 16.63 6.4 17.066 6.313 17.415 6.487 C 17.763 6.575 18.025 7.011 17.938 7.36 L 17.153 13.382 L 24.041 13.382 Z M 17.066 17.832 L 16.543 21.847 L 22.123 15.127 L 16.107 15.127 C 15.932 15.127 15.671 15.039 15.496 14.865 C 15.409 14.69 15.322 14.428 15.322 14.167 L 15.758 10.152 L 10.178 16.872 L 16.194 16.872 C 16.455 16.872 16.717 16.959 16.892 17.134 C 16.979 17.308 17.066 17.57 17.066 17.832 Z"
                fill="white"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1m5binv-MuiTypography-root"
            >
              All-in-one zap contracts
            </p>
          </div>
          <div
            class="ZapInfo-infoBody MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 oly-info-body-header css-69sme-MuiTypography-root"
            >
              All-in-one easy staking
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft oly-info-body css-z19mi8-MuiTypography-root"
            >
              OlyZap reduces complexity, saves you time and keeps you here on Olympus.
            </p>
          </div>
        </div>
        <div
          class="ZapInfo-infoBox MuiGrid-root ZapInfo-infoBoxItem MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-c8aaa2-MuiGrid-root"
        >
          <div
            class="ZapInfo-infoHeader oly-info-header-box MuiBox-root css-1m5be6d"
          >
            <div
              class="MuiBox-root css-1xhj18k"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1djwm8g-MuiSvgIcon-root"
                focusable="false"
                style="z-index: 1; margin-bottom: 16px;"
                viewBox="0 0 32 32"
              >
                <defs>
                  <lineargradient
                    gradientTransform="matrix(0.319996, 0, 0, 0.319996, -0.000007, -1.775952)"
                    gradientUnits="userSpaceOnUse"
                    id="paint0_linear"
                    x1="-83.5339"
                    x2="189.059"
                    y1="50"
                    y2="50"
                  >
                    <stop
                      stop-color="#708B96"
                    />
                    <stop
                      offset="1"
                      stop-color="#F7FBE7"
                    />
                  </lineargradient>
                </defs>
                <path
                  clip-rule="evenodd"
                  d="M 15.982 28.228 C 23.688 28.228 29.934 21.982 29.934 14.276 C 29.934 12.658 29.658 11.104 29.152 9.659 L 29.152 5.11 C 30.947 7.696 32 10.837 32 14.224 C 32 23.06 24.836 30.224 16 30.224 C 7.163 30.224 0 23.06 0 14.224 C 0 10.817 1.065 7.659 2.88 5.064 L 2.88 9.472 C 2.331 10.97 2.031 12.588 2.031 14.276 C 2.031 21.982 8.276 28.228 15.982 28.228 Z"
                  fill="url(#paint0_linear)"
                  fill-rule="evenodd"
                />
                <path
                  d="M 3.552 14.224 C 3.552 7.349 9.125 1.776 16 1.776 C 22.875 1.776 28.448 7.349 28.448 14.224 C 28.448 21.099 22.875 26.671 16 26.671 C 9.125 26.671 3.552 21.099 3.552 14.224 Z"
                  fill="#708B96"
                />
                <path
                  clip-rule="evenodd"
                  d="M 17.195 19.176 C 17.195 19.176 17.195 19.176 17.195 19.176 L 17.195 19.726 L 21.477 19.726 L 21.477 18.181 L 19.363 18.181 C 20.634 17.219 21.451 15.72 21.451 14.037 C 21.451 11.129 19.011 8.772 16 8.772 C 12.989 8.772 10.548 11.129 10.548 14.037 C 10.548 15.72 11.365 17.219 12.637 18.181 L 10.523 18.181 L 10.523 19.726 L 14.805 19.726 L 14.805 19.176 L 14.805 17.652 C 13.245 17.16 12.116 15.733 12.116 14.05 C 12.116 11.96 13.855 10.266 16 10.266 C 18.145 10.266 19.884 11.96 19.884 14.05 C 19.884 15.733 18.755 17.16 17.195 17.652 L 17.195 19.176 Z"
                  fill="white"
                  fill-rule="evenodd"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1djwm8g-MuiSvgIcon-root"
                focusable="false"
                style="margin-left: -6px; z-index: 1; margin-bottom: 16px;"
                viewBox="0 0 32 32"
              >
                <defs>
                  <lineargradient
                    gradientTransform="matrix(0.177778, 0, 0, 0.177778, 1.777779, 1.777779)"
                    gradientUnits="userSpaceOnUse"
                    id="paint0_linear_359"
                    x1="80"
                    x2="80"
                    y1="-84"
                    y2="256"
                  >
                    <stop
                      offset="0.1949"
                      stop-color="#708B96"
                    />
                    <stop
                      offset="1"
                      stop-color="#F7FBE7"
                    />
                  </lineargradient>
                </defs>
                <path
                  d="M 1.778 16 C 1.778 8.145 8.145 1.778 16 1.778 C 23.855 1.778 30.222 8.145 30.222 16 C 30.222 23.855 23.855 30.222 16 30.222 C 8.145 30.222 1.778 23.855 1.778 16 Z"
                  fill="#fff"
                />
                <rect
                  fill="#fff"
                  height="12.516"
                  width="12.516"
                  x="9.742"
                  y="9.771"
                />
                <path
                  clip-rule="evenodd"
                  d="M 14.635 22.286 L 14.635 19.916 C 12.852 19.355 11.563 17.725 11.563 15.801 C 11.563 13.413 13.549 11.477 16 11.477 C 18.451 11.477 20.437 13.413 20.437 15.801 C 20.437 17.725 19.148 19.355 17.365 19.916 L 17.365 21.658 L 17.365 22.258 L 17.365 22.286 L 22.258 22.286 L 22.258 20.523 L 19.842 20.523 C 21.295 19.421 22.229 17.709 22.229 15.787 C 22.229 12.464 19.44 9.771 16 9.771 C 12.56 9.771 9.771 12.464 9.771 15.787 C 9.771 17.709 10.705 19.421 12.158 20.523 L 9.742 20.523 L 9.742 22.286 Z"
                  fill="#708b96"
                  fill-rule="evenodd"
                />
                <path
                  d="M 16 28.444 C 9.127 28.444 3.556 22.873 3.556 16 L 0 16 C 0 24.837 7.163 32 16 32 Z M 28.444 16 C 28.444 22.873 22.873 28.444 16 28.444 L 16 32 C 24.837 32 32 24.837 32 16 Z M 16 3.556 C 22.873 3.556 28.444 9.127 28.444 16 L 32 16 C 32 7.163 24.837 0 16 0 Z M 16 0 C 7.163 0 0 7.163 0 16 L 3.556 16 C 3.556 9.127 9.127 3.556 16 3.556 Z"
                  fill="url(#paint0_linear_359)"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1m5binv-MuiTypography-root"
            >
              You Choose
            </p>
          </div>
          <div
            class="ZapInfo-infoBody MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 oly-info-body-header css-69sme-MuiTypography-root"
            >
              Staking
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft oly-info-body css-z19mi8-MuiTypography-root"
            >
              Staking is the primary value accrual strategy of Olympus. When you stake, you lock OHM and receive an equal value of sOHM or gOHM.
            </p>
          </div>
        </div>
      </div>
      <div
        class="button-box MuiBox-root css-0"
      >
        <a
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root learn-more-button css-1sigo3z-MuiButtonBase-root-MuiButton-root"
          href="https://docs.olympusdao.finance/main/using-the-website/olyzaps"
          tabindex="0"
          target="_blank"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-69sme-MuiTypography-root"
          >
            Learn More
          </p>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1x7u7wj-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 20 20"
          >
            arrow-up.svg
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
